### PR TITLE
Alusine-Node.js-homework-week1

### PR DESCRIPTION
**Pull request by alquinex**

### CODE REVIEW

File: week1/homework/src/server.js [hunk 1]
```diff
@@ -8,8 +8,37 @@ const http = require('http');
 function createServer(port) {
   let state = 10;
 
+  function displayResponse(response, state) {
```
Comment (gajduk) [inline week1/homework/src/server.js]: Love that you separated and reused this code 🥇

File: week1/homework/src/server.js [hunk 2]
```diff
@@ -8,8 +8,37 @@ const http = require('http');
...
+    response.setHeader('Content-Type', 'application/json');
+    response.write(JSON.stringify(state));
+    response.end();
+  }
+
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        displayResponse(response, { state });
```
Comment (gajduk) [inline week1/homework/src/server.js]: very tidy 👍

File: week1/homework/src/server.js [hunk 3]
```diff
@@ -8,8 +8,37 @@ const http = require('http');
...
+        break;
+
+      case '/reset':
+        state = 10;
+        displayResponse(response, { state });
+        break;
+
+      default:
+        response.statusCode = 404;
```
Comment (gajduk) [inline week1/homework/src/server.js]: 404 👍

File: week1/homework/src/server.js [hunk 4]
```diff
@@ -8,8 +8,37 @@ const http = require('http');
...
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        displayResponse(response, { state });
+        break;
+
+      case '/add':
+        state++;
+        displayResponse(response, { state });
```
Comment (gajduk) [inline week1/homework/src/server.js]: honestly had to look up how `{ state }` works. I like it much more than  `{ 'state': state } ` Very good 👍

**Comment by alquinex** [comment]
Hi @gajduk, 
Thanks for the feedback.

**Pull request closed without merging**